botcity/core/application.py

```diff
@@ -0,0 +1,53 @@
+import enum
+import platform
+from functools import wraps
+
+
+def if_windows_os(func):
+    """
+    Decorator which raises if the OS is not Windows.
+
+    Args:
+        func (callable): The function to be wrapped
+
+    Returns:
+        wrapper (callable): The decorated function
+    """
+    @wraps(func)
+    def wrapper(self, *args, **kwargs):
+        if platform.system() != "Windows":
```

Comment:
What do you think about changing the if negation to something more solid? For example:
```suggestion
if plataform.system() == "Windows":
    return func(self, *args, **kwargs)
raise ValueError(f'You can connect to an application on Windows OS only. Cannot invoke {func.__name__}.')
```
That way the name of the function is more objective. What do you think of this proposal?

Reply:
@kayqueGovetri good observations! I was thinking of including the methods in the `application.py` I had created, leaving it along with the Backend enum and validation decorators. That way, everything would be concentrated in the same 'application' module.

Do you think it would be a good strategy? Or would it be more interesting to leave the methods separate?

Reply:
@joao-voltarelli It's a good alternative to make a preview of how it would look.

Something I could do to make it more isolated would be to create sub levels for each situation, for example:

- application.py can become a package application/
- Inside the package it can contain:
   - The enums.
   - The decorators.
   - The functions.
   - A `___init__`.py file containing the `__all__` method to make exporting files easier.

What do you think of this approach? It might be more work but I think it's worth the result.

Reply:
It was an alternative that I was considering as well, it would make it easier to include future implementations of functions, decorators...
I'll take a look at these options and see how it gets better organized.